cli/bpmetadata/types.go

```diff
@@ -4,92 +4,141 @@ import (
 }
 
 type BlueprintDescription struct {
-	Tagline      string   `json:",omitempty" yaml:",omitempty"`
-	Detailed     string   `json:",omitempty" yaml:",omitempty"`
-	PreDeploy    string   `json:"preDeploy,omitempty" yaml:"preDeploy,omitempty"`
+	// Autogenerated: All types of descriptions are set with the markdown content
+	// immediately after the each type's heading declaration in readme.md.
+
+	// Autogenerated: Markdown after "### Tagline"
+	Tagline string `json:",omitempty" yaml:",omitempty"`
+	// Autogenerated: Markdown after "### Detailed"
+	Detailed string `json:",omitempty" yaml:",omitempty"`
+	// Autogenerated: Markdown after "### PreDeploy"
+	PreDeploy string `json:"preDeploy,omitempty" yaml:"preDeploy,omitempty"`
+	// Autogenerated: Markdown after "### Html"
+	Html string `json:"html,omitempty" yaml:"html,omitempty"`
+	// Autogenerated: Markdown after "### EulaUrls"
+	EulaUrls []string `json:"eulaUrls,omitempty" yaml:"eulaUrls,omitempty"`
+	// Autogenerated: Markdown after "### Architecture"
+	// Deprecated. Use BlueprintContent.Architecture instead.
 	Architecture []string `json:"architecture,omitempty" yaml:"architecture,omitempty"`
 }
 
+// A time estimate in secs required for configuring the solution
 type BlueprintTimeEstimate struct {
+	// Autogenerated: Set using the content defined under "### DeploymentTime" E.g.
+	// ### DeploymentTime
+	// - Configuration: X secs
+	// - Deployment: Y secs
 	ConfigurationSecs int `json:"configuration,omitempty" yaml:"configuration,omitempty"`
 	DeploymentSecs    int `json:"deployment,omitempty" yaml:"deployment,omitempty"`
 }
 
+// The cost estimate for the solution based on preconfigured variables.
 type BlueprintCostEstimate struct {
+	// Autogenerated: Set using the content defined under "### Cost" as a link
+	// with a description E.g.
+	// ### Cost
+	// [View cost details](https://cloud.google.com/products/calculator?hl=en_US&_ga=2.1665458.-226505189.1675191136#id=02fb0c45-cc29-4567-8cc6-f72ac9024add)
 	Description string `json:",omitempty" yaml:",omitempty"`
 	Url         string `json:",omitempty" yaml:",omitempty"`
 }
 
+// A GCP cloud product used in the solution.
+// Manually authored.
 type BlueprintCloudProduct struct {
-	ProductId   string `json:",omitempty" yaml:",omitempty"`
-	PageUrl     string `json:",omitempty" yaml:",omitempty"`
-	Label       string `json:",omitempty" yaml:",omitempty"`
-	LocationKey bool   `json:",omitempty" yaml:",omitempty"`
+	// A top-level (e.g. "Compute Engine") or secondary (e.g. "Binary Authorization")
+	// product used in the solution.
+	ProductId string `json:",omitempty" yaml:",omitempty"`
+	// Url for the product
+	PageUrl string `json:",omitempty" yaml:",omitempty"`
+	// A label string for the product, if it is not an integrated GCP product.
+	// E.g. "Data Studio"
+	Label string `json:",omitempty" yaml:",omitempty"`
+	// If this product is a key in determining the GCP regions/zones the solution
```

Comment:
It means that this product is limited to certain regions and it should be a "key" in determine the allowed locations for the solution. Although, now that I think about it, I don't think this should be field since we should be able to determine that from the product Id implicitly. I'm going to take this out.